### DISCUSSION TITLE
Question about removing unused css

### DISCUSSION THREAD

**Discussion by tankpower1** [vercel/next.js#10682] [Help] [answered]
Hello,
I wanted to ask how can one remove all the unused css in his code using Next.js.

My project uses scss and has 1 starting point (app.scss) which imports all other partial .scss files.
In next.config.js i'm using withSass and i've tried using withPurgeCss but it removes all css instead of the unused one.

**Comment by Timer** [comment, marked as answer]
If you upgrade to the latest Next.js canary we support Sass out of the box. Be sure you delete `@zeit/next-sass`!

As for eliminating CSS, be sure you've included all relevant files. See our tailwind example for a known working setup:
https://github.com/zeit/next.js/blob/canary/examples/with-tailwindcss/postcss.config.js

**Reply by starmatt** [reply]
What if you don't use tailwind? It seems from your example that tailwind does the purging.
Is there a way to do so with PurgeCSS and Next.js built-in css feature?

Thanks!

**Reply by dpkpaa** [reply]
same here

**Reply by Manikanta-20** [reply]
@Timer 
next-purgecss  will work on 
1. next-css
2. next-less
3. next-sass
?


I'm not using anyone of these next-(css/less/sass).using only **material ui & styled-components ,in this case how to remove the unused css**?

**Reply by moneszarrugh** [reply]
I also would like to remove unused css of bootstrap. Why is there only example of that for tailwind?

**Reply by ahmadawais** [reply]
@Timer I want to remove unused CSS on a page to page basis. Do you have a recommendation there? E.g. Blog posts use Tailwind and the Homepage doesn't. The homepage ends up with Tailwind CSS that's not used and excessive. Since I have to use `_app.js` for CSS and can't put it inside the blog posts layout template — I'm kind of stuck with this CSS extra baggage.

**Reply by Pfurr** [reply]
@ahmadawais @Timer  same! Any idea?

**Reply by ahmadawais** [reply]
I have no idea how to do this. I hope @timer or Tim (Next team) would make this possible some how.
